place_tracker/lib/place_tracker_app.dart

```diff
@@ -21,10 +23,29 @@ class PlaceTrackerApp extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return const MaterialApp(
-      home: _PlaceTrackerHomePage(),
+    return MaterialApp.router(
+      routerConfig: GoRouter(routes: [
+        GoRoute(
+          path: '/',
+          builder: (context, state) => const _PlaceTrackerHomePage(),
+          routes: [
+            GoRoute(
+              path: 'place/:id',
+              builder: (context, state) {
+                final id = state.params['id']!;
+                final place = context
+                    .read<AppState>()
+                    .places
+                    .singleWhere((place) => place.id == id);
+                return PlaceDetails(place: place);
```

Comment:
Possible improvement here, handle when the provided ID does not exist

Reply:
That's a good call-out -  I think this is okay as written, assuming the app doesn't link to any invalid IDs.